airflow/providers/databricks/operators/databricks.py

```diff
@@ -511,6 +512,11 @@ def __init__(
         self.databricks_retry_delay = databricks_retry_delay
         self.wait_for_termination = wait_for_termination
 
+        if job_name or 'job_name' in self.json:
```

Comment:
You can move this after adding job_name and job_id to the json to simplify the code and the checks:
```
if job_name:
    self.json['job_name'] = job_name
if job_id is not None:
    self.json['job_id'] = job_id
if 'job_name' in self.json and 'job_id' in self.json:
    raise AirflowException("Argument 'job_name' is not allowed with argument 'job_id'")
```

Reply:
totally agree, better to keep it in json, changed it